### ISSUE TITLE
VSCode sync with local installation

### ISSUE THREAD

**Issue by Yajo** [gitpod-io/gitpod#3733]
I'd love a way to sync settings from gitpod’s VSCode to my local VSCode installed in my laptop. That would provide an even more familiar workspace for me, with all the snippets, keybindings and extensions I’m used to.

**Comment by svenefftinge** [comment]
VS Code currently sets the user sync server provider in the product.json. 
So we would need to alter that somehow. Our local app could possibly do that.

**Comment by akosyakov** [comment]
There is a hidden config, but i am not sure that GitHub tokens acuired by MS github authentication extension will work for authentication with us.

**Comment by akosyakov** [comment]
We will implement Gitpod auth as a part of the Gitpod extension for VS Code Desktop, and distribute this extension as built-in for VSCode Web Server in order to resolve https://github.com/gitpod-io/vscode/issues/29 as well.

**Comment by roboquat** [comment]
@akosyakov: Issue scheduled in the IDE team (WIP: 0)

<details>

In response to [this](https://github.com/gitpod-io/gitpod/issues/3733#issuecomment-917884570):

>/schedule


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by akosyakov** [comment]
I assigned myself for now, but @filiptronicek is working on it.

**Comment by akosyakov** [comment]
@mikenikles This one is tricky in the sense of CHANGELOG, the PR is done against another repo. So there is no automation.

**Comment by julkhami** [comment]
I’d like to be able to keep the same layout in my VS code when I reopen GitPod and someone directed me to this thread.
Is it currently possible, if I drag closed the file explorer, to save those settings? Or is this feature ongoing? Thanks very much

**Comment by akosyakov** [comment]
> I’d like to be able to keep the same layout in my VS code when I reopen GitPod and someone directed me to this thread.
Is it currently possible, if I drag closed the file explorer, to save those settings? Or is this feature ongoing? Thanks very much

This issue is about auto synching with VS Code Desktop settings. Settings Sync in VS Code Web already enabled by default. Please consult VS Code settings docs for possible layout customizations.

**Comment by schickling** [comment]
As I keep my VSC [`settings.json` in my dotfiles](https://github.com/schickling/dotfiles/blob/main/VSCode/settings.json), it would be great if I could [link to it in the dotfiles install script](https://github.com/schickling/dotfiles/blob/aa07dd2c24503eac5c0d1fe3f6973cb3100159ad/install.sh#L31-L33).

**Comment by Yajo** [comment]
Thanks @rrapstine. I currently use that extension locally because I recently switched to VSCodium. However, it seems forbidden in gitpod:

![imagen](https://user-images.githubusercontent.com/973709/154036262-2397eeac-86f2-48f9-af2d-9d197364f82b.png)

**Comment by loujaybee** [comment]
Removed epic label from this issue in favour of: https://github.com/gitpod-io/gitpod/issues/9136

**Comment by schickling** [comment]
[I keep my VSC settings files manually in my dotfiles repo](https://github.com/schickling/dotfiles/tree/main/VSCode). Is there a way to use my VSC settings files in Gitpod instead of the automatic settings sync?

**Comment by jeanp413** [comment]
Not possible as settings in vscode browser are stored in the browser localstorage so only way to sync them is using settings sync

**Comment by schickling** [comment]
> Not possible as settings in vscode browser are stored in the browser localstorage so only way to sync them is using settings sync

Wouldn't it be possible for Gitpod to support some kind of "import settings from JSON" workflow? (e.g. via the `gp` CLI?)

Can I use both VSC sync (via Microsoft) and VSC sync (via Gitpod) at the same time?

**Comment by jeanp413** [comment]
> Wouldn't it be possible for Gitpod to support some kind of "import settings from JSON" workflow? (e.g. via the gp CLI?)

Maybe a command in the gitpod extension that reads a file and uses the vscode API to update settings could work, but what about merge conflicts, etc. I suggest you to create a seprate issue as a feature request.

> Can I use both VSC sync (via Microsoft) and VSC sync (via Gitpod) at the same time?

Nope, why would you want to use both? Are you using codespaces or vscode.dev too? (we plan to investigate if our settings sync also works there)

**Comment by schickling** [comment]
> Nope, why would you want to use both? Are you using codespaces or vscode.dev too? (we plan to investigate if our settings sync also works there)

Yes, I'm using `github.dev` a lot as well. I don't want to have to choose for one or the other.
